aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -1094,13 +1094,31 @@ Tensor rand_like(
     std::optional<Device> device,
     std::optional<bool> pin_memory,
     std::optional<c10::MemoryFormat> optional_memory_format) {
+  return native::rand_like(
+      self,
+      static_cast<std::optional<Generator>>(std::nullopt),
+      dtype,
+      layout,
+      device,
+      pin_memory,
+      optional_memory_format);
+}
+
+Tensor rand_like(
+    const Tensor& self,
+    std::optional<Generator> generator,
+    std::optional<ScalarType> dtype,
```

Comment:
Is there a reason we keep two interfaces and not a single one with an argument by default?

Reply:
I followed the example of `rand`, which has separate implementations for (I assume) backward compatibility. I don’t see any problematic calls to these functions but I’m still getting to know the codebase so I can’t fully assess the implications of removing the original signatures.

Also sorry for the noise everyone—a bad rebase (since resolved) created spurious commits and triggered unnecessary review requests that I don’t have permission to undo.